controllers/controllers/workloads/cforg_controller_test.go

```diff
@@ -0,0 +1,415 @@
+					_, patchedHierarchy, _, _ := fakeClient.PatchArgsForCall(0)
+					castHierarchy, ok := patchedHierarchy.(*v1alpha2.HierarchyConfiguration)
+					Expect(ok).To(BeTrue(), "Cast to v1alpha2.HierarchyConfiguration failed")
+					Expect(castHierarchy.Spec.AllowCascadingDeletion).To(BeTrue())
+
+					// validate the inputs to Status.Update
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(2))
+					_, updatedCFOrg, _ := fakeStatusWriter.UpdateArgsForCall(1)
+					castCFOrg, ok := updatedCFOrg.(*workloadsv1alpha1.CFOrg)
+					Expect(ok).To(BeTrue(), "Cast to workloadsv1alpha1.CFOrg failed")
+					Expect(meta.IsStatusConditionTrue(castCFOrg.Status.Conditions, StatusConditionReady)).To(BeTrue(), "Status Condition "+StatusConditionReady+" was not True as expected")
+					Expect(castCFOrg.Status.GUID).To(Equal(namespace.Name))
+				})
+			})
+
+			When("a subnamespace anchor exists", func() {
+				BeforeEach(func() {
+					subNamespaceAnchorError = nil
+					setSubNamespaceStatus = false
+					subNamespaceAnchor.Status.State = v1alpha2.Ok
+				})
+
+				It("returns an empty result and does not return error", func() {
+					Expect(reconcileResult).To(Equal(ctrl.Result{}))
+					Expect(reconcileErr).NotTo(HaveOccurred())
+
+					// validate the inputs to Get
+					Expect(fakeClient.GetCallCount()).To(Equal(3))
+					_, namespacedName, _ := fakeClient.GetArgsForCall(0)
+					Expect(namespacedName.Namespace).To(Equal(defaultNamespace))
+					Expect(namespacedName.Name).To(Equal(cfOrgGUID))
+					_, namespacedName, _ = fakeClient.GetArgsForCall(1)
+					Expect(namespacedName.Namespace).To(Equal(defaultNamespace))
+					Expect(namespacedName.Name).To(Equal(cfOrgGUID))
+					_, namespacedName, _ = fakeClient.GetArgsForCall(2)
+					Expect(namespacedName.Name).To(Equal(cfOrgGUID))
+
+					// validate Create is not called
+					Expect(fakeClient.CreateCallCount()).To(Equal(0))
+
+					// validate the inputs to Patch
+					Expect(fakeClient.PatchCallCount()).To(Equal(1))
+					_, patchedHierarchy, _, _ := fakeClient.PatchArgsForCall(0)
+					castHierarchy, ok := patchedHierarchy.(*v1alpha2.HierarchyConfiguration)
+					Expect(ok).To(BeTrue(), "Cast to v1alpha2.HierarchyConfiguration failed")
+					Expect(castHierarchy.Spec.AllowCascadingDeletion).To(BeTrue())
+
+					// validate the inputs to Status.Update
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(1))
+					_, updatedCFOrg, _ := fakeStatusWriter.UpdateArgsForCall(0)
+					castCFOrg, ok := updatedCFOrg.(*workloadsv1alpha1.CFOrg)
+					Expect(ok).To(BeTrue(), "Cast to workloadsv1alpha1.CFOrg failed")
+					Expect(meta.IsStatusConditionTrue(castCFOrg.Status.Conditions, StatusConditionReady)).To(BeTrue(), "Status Condition "+StatusConditionReady+" was not True as expected")
+					Expect(castCFOrg.Status.GUID).To(Equal(namespace.Name))
+				})
+			})
+		})
+
+		When("on the unhappy path", func() {
+			When("fetch CFOrg returns an error", func() {
```

Comment:
On the cases that there are errors, it is essential to check that an error is returned and there were no unnecessary `create/update` calls. I don't think the repeating all the checks is necessary, e.g do you care that get has been invoked (this has been tested already above)?

Reply:
Done